t/t7900-maintenance.sh

```diff
@@ -1012,4 +1012,129 @@ test_expect_success 'repacking loose objects is quiet' '
 	)
 '
 
+test_expect_success 'vfs-cache-move task with no shared cache no op' '
+	test_when_finished "rm -rf repo" &&
+	git init repo &&
+	(
+		cd repo &&
+
+		test_commit something &&
+		git config set maintenance.gc.enabled false &&
+		git config set maintenance.vfs-cache-move.enabled true &&
+		git config set maintenance.vfs-cache-move.auto 1 &&
+
+		printf "blob\ndata <<END\n%s\nEND\n\n" 1 2 3 4 5 | \
+		git -c fastimport.unpackLimit=0 fast-import &&
+		find .git/objects/pack \
+			-type f \
+			\( -name "*.pack" \
+			-o -name "*.idx" \
+			-o -name "*.keep" \
+			-o -name "*.rev" \) >files.txt &&
+
+		git maintenance run &&
+		while IFS= read -r f; do
+			test_path_exists $f || exit 1
+		done <files.txt
+	)
+'
+
+test_expect_success 'vfs-cache-move task cache path same as pack dir no op' '
+	test_when_finished "rm -rf repo" &&
+	git init repo &&
+	(
+		cd repo &&
+
+		test_commit something &&
+		git config set gvfs.sharedcache .git/objects/pack &&
+		git config set maintenance.gc.enabled false &&
+		git config set maintenance.vfs-cache-move.enabled true &&
+		git config set maintenance.vfs-cache-move.auto 1 &&
+
+		printf "blob\ndata <<END\n%s\nEND\n\n" 1 2 3 4 5 | \
+		git -c fastimport.unpackLimit=0 fast-import &&
+		find .git/objects/pack \
+			-type f \
+			\( -name "*.pack" \
+			-o -name "*.idx" \
+			-o -name "*.keep" \
+			-o -name "*.rev" \) >files.txt &&
+
+		git maintenance run &&
+		while IFS= read -r f; do
+			test_path_exists $f || exit 1
+		done <files.txt
+	)
+'
+
+test_expect_success 'vfs-cache-move task no .rev or .keep' '
+	test_when_finished "rm -rf repo cache" &&
+	mkdir -p cache/pack &&
+	git init repo &&
+	(
+		cd repo &&
+
+		test_commit something &&
+		git config set gvfs.sharedcache ../cache &&
+		git config set maintenance.gc.enabled false &&
+		git config set maintenance.vfs-cache-move.enabled true &&
+		git config set maintenance.vfs-cache-move.auto 1 &&
+
+		printf "blob\ndata <<END\n%s\nEND\n\n" 1 2 3 4 5 | \
+		git -c fastimport.unpackLimit=0 fast-import &&
+		find .git/objects/pack \
+			-type f \
+			\( -name "*.pack" \
+			-o -name "*.idx" \) >src.txt &&
+
+		rm -f .git/objects/pack/*.rev .git/objects/pack/*.keep &&
+
+		sed "s|.*/|../cache/pack/|" src.txt >dst.txt &&
```

Comment:
Isn't `s|...|...|` a GNU extension? I.e. does this work on macOS? From https://pubs.opengroup.org/onlinepubs/9699919799/utilities/sed.html it looks to me as if only `s/.../.../` would be portable...